### PR TITLE
cleanup network manager changes

### PR DESCRIPTION
**Pull request by StephenHodgson**
- removed debug
- reverted misc changes from upstream

### CODE REVIEW

File: com.unity.netcode.gameobjects/Runtime/SceneManagement/NetworkSceneManager.cs [hunk 1]
```diff
@@ -2078,7 +2077,6 @@ internal void SynchronizeNetworkObjects(ulong clientId)
                     {
                         continue;
                     }
-
```
Comment (afarchy) [inline com.unity.netcode.gameobjects/Runtime/SceneManagement/NetworkSceneManager.cs:2081]: this is stupid to debate but I really don't like code on the line after `}`

File: com.unity.netcode.gameobjects/Runtime/SceneManagement/NetworkSceneManager.cs [hunk 2]
```diff
@@ -776,15 +774,16 @@ internal string ScenePathFromHash(uint sceneHash)
...
-#endif
+            var addressableKey = UnityEditor.AssetDatabase.AssetPathToGUID(sceneNameOrPath);
+
             if (AddressableKeyToHash.TryGetValue(addressableKey, out var externalSceneHash))
             {
-                Debug.Log($"SceneHashFromNameOrPath: {addressableKey} = {externalSceneHash}");
+                return externalSceneHash;
+            }
+#endif
+            if (AddressableKeyToHash.TryGetValue(sceneNameOrPath, out externalSceneHash))
```
Comment (afarchy) [inline com.unity.netcode.gameobjects/Runtime/SceneManagement/NetworkSceneManager.cs]: revert this please, there's more work you'd have to do if you want to support looking up by scene name. We don't store the scene name in this hashmap.

**Review by afarchy — DISMISSED** [review]

File: com.unity.netcode.gameobjects/Runtime/SceneManagement/NetworkSceneManager.cs (hunk 2 above)
Comment (afarchy) [reply com.unity.netcode.gameobjects/Runtime/SceneManagement/NetworkSceneManager.cs]: and I don't want more regressions in this area

File: com.unity.netcode.gameobjects/Runtime/SceneManagement/NetworkSceneManager.cs (hunk 1 above)
Comment (StephenHodgson) [reply com.unity.netcode.gameobjects/Runtime/SceneManagement/NetworkSceneManager.cs:2081]: Well I would rather it be "exactly" the same as upstream. So prefer to keep the diff smaller so we don't have conflicts if we have to maintain fork.

File: com.unity.netcode.gameobjects/Runtime/SceneManagement/NetworkSceneManager.cs (hunk 2 above)
Comment (StephenHodgson) [reply com.unity.netcode.gameobjects/Runtime/SceneManagement/NetworkSceneManager.cs]: I'll revert then, but would prefer that we support both.

**Pull request merged**